src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -114,42 +116,49 @@ public synchronized void init() {
-        // Have Mesos fill in the current user.
-        FrameworkInfo framework = FrameworkInfo.newBuilder()
-          .setUser(targetUser == null ? "" : targetUser)
-          .setName(mesosCloud.getFrameworkName())
-          .setPrincipal(mesosCloud.getPrincipal())
-          .setCheckpoint(mesosCloud.isCheckpoint())
-          .setWebuiUrl(webUrl != null ? webUrl :  "")
-          .build();
-
-        LOGGER.info("Initializing the Mesos driver with options"
-        + "\n" + "Framework Name: " + framework.getName()
-        + "\n" + "Principal: " + framework.getPrincipal()
-        + "\n" + "Checkpointing: " + framework.getCheckpoint()
-        );
-
-        if (StringUtils.isNotBlank(mesosCloud.getSecret())) {
+        try {
+          String targetUser = mesosCloud.getSlavesUser();
+          String webUrl = Jenkins.getInstance().getRootUrl();
+          if (webUrl == null) webUrl = System.getenv("JENKINS_URL");
+          StandardUsernamePasswordCredentials credentials = mesosCloud.getCredentials();
+          String principal = credentials == null ? "" : credentials.getUsername();
+          String secret = credentials == null ? "" : Secret.toString(credentials.getPassword());
+          // Have Mesos fill in the current user.
+          FrameworkInfo framework = FrameworkInfo.newBuilder()
+            .setUser(targetUser == null ? "" : targetUser)
+            .setName(mesosCloud.getFrameworkName())
+            .setPrincipal(principal)
+            .setCheckpoint(mesosCloud.isCheckpoint())
+            .setWebuiUrl(webUrl != null ? webUrl : "")
+            .build();
+
+          LOGGER.info("Initializing the Mesos driver with options"
+            + "\n" + "Framework Name: " + framework.getName()
+            + "\n" + "Principal: " + principal
+            + "\n" + "Checkpointing: " + framework.getCheckpoint()
+          );
+
+          if (StringUtils.isNotBlank(secret)) {
             Credential credential = Credential.newBuilder()
-              .setPrincipal(mesosCloud.getPrincipal())
-              .setSecret(ByteString.copyFromUtf8(mesosCloud.getSecret()))
+              .setPrincipal(principal)
+              .setSecret(ByteString.copyFromUtf8(secret))
               .build();
 
-            LOGGER.info("Authenticating with Mesos master with principal " + credential.getPrincipal());
+            LOGGER.info("Authenticating with Mesos master with principal " + principal);
             driver = new MesosSchedulerDriver(JenkinsScheduler.this, framework, mesosCloud.getMaster(), credential);
-        } else {
+          } else {
             driver = new MesosSchedulerDriver(JenkinsScheduler.this, framework, mesosCloud.getMaster());
+          }
+          Status runStatus = driver.run();
+          if (runStatus != Status.DRIVER_STOPPED) {
+            LOGGER.severe("The Mesos driver was aborted! Status code: " + runStatus.getNumber());
+          }
+        } catch (RuntimeException e) {
```

Comment:
why is a "try catch" changed sneaked in here? If it's not related to this change, please split that out into its own PR.